### PR TITLE
Fix pyarrow mkdirs

### LINKED ISSUE #6467: write_parquet using partition_on breaking on GCS
**What happened**:

Saving a partitioned dask dataframe on google cloud storage raises the following error

```shell
/usr/local/lib/python3.7/site-packages/dask/dataframe/io/parquet/arrow.py in _write_partitioned()
    246         )
    247         prefix = fs.sep.join([root_path, subdir])
--> 248         fs.mkdir(prefix, exists_ok=True)
    249         outfile = guid() + ".parquet"
    250         full_path = fs.sep.join([prefix, outfile])

TypeError: mkdir() got an unexpected keyword argument 'exists_ok'
```

**What you expected to happen**:

The file should be saved on GCS using a folder structure for the partitions.

**Minimal Complete Verifiable Example**:

```python
import dask.dataframe as dd
dd.from_pandas(pd.DataFrame({'foo': [1, 1, 2, 2]}), npartitions=1).to_parquet('gs://...', partition_on=['foo'])
```

**Anything else we need to know?**:

1. Locally it works fine
2. Maybe there is a typo on https://github.com/dask/dask/blob/05c17758aa9d44158538b7d5059524de18843ae3/dask/dataframe/io/parquet/arrow.py#L75

**Environment**:

- Dask version: 2.21.0
- GCSFS version: 0.6.2
- Python version: 3.7.6
- Operating System: Ubuntu 18.04
- Install method (conda, pip, source): pip

### PR DESCRIPTION
**Pull request by jsignell**
- [ ] Tests added / passed
- [x] Passes `black dask` / `flake8 dask`

Closes #6467

### CODE REVIEW

File: dask/dataframe/io/parquet/arrow.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def _write_partitioned(table, root_path, partition_cols, fs, index_cols=(), **kw
             subgroup, preserve_index=preserve_index, schema=subschema, safe=False
         )
         prefix = fs.sep.join([root_path, subdir])
-        fs.mkdir(prefix, exists_ok=True)
+        fs.mkdirs(prefix, exists_ok=True)
```
Comment (hermesribeiro) [inline dask/dataframe/io/parquet/arrow.py:75]: Should the second argument be `exist_ok=True`?

Comment (jsignell) [reply dask/dataframe/io/parquet/arrow.py:75]: haha I didn't realize there were two different parts wrong :)

Comment (jsignell) [reply dask/dataframe/io/parquet/arrow.py:75]: I guess that means I should write a test...

Comment (jsignell) [reply dask/dataframe/io/parquet/arrow.py:75]: I was being silly. It is already failing tests :facepalm:

Comment (hermesribeiro) [reply dask/dataframe/io/parquet/arrow.py]: Good! Didn't check the logs, but it's very nice that it covered it already.

**Comment by jsignell** [comment]
cc @rjzamora for a sanity check that this is right.

**Pull request merged**